functions/controllers/applicants/index.js

```diff
@@ -0,0 +1,42 @@
+const functions = require("firebase-functions");
+const express = require("express");
+const cors = require("cors");
+const { jwtCheck, hasReadAdmin } = require("../../utils/middleware");
+const { queryDocument } = require("../../utils/database");
+
+const applicant = express();
+applicant.disable("x-powered-by");
+applicant.use(express.json());
+
+const auth0Config = functions.config().auth;
+const corsConfig = auth0Config ? auth0Config.cors : "";
+
+const corsOptions = {
+  origin: corsConfig,
+  optionsSuccessStatus: 200,
+};
+
+applicant.use(cors(corsOptions));
+
+applicant.get("/applicant/:id/", jwtCheck, hasReadAdmin, async (req, res) => {
+  try {
+    const snapshot = await queryDocument("applicants", req.params.id);
+    if (!snapshot.exists) {
+      throw new Error("Applicant doesn't exist for the given ID");
+    }
+    return res.status(200).send({
+      error: false,
+      status: 200,
+      message: snapshot.data(),
+    });
+  } catch (error) {
+    if (error.message === "Applicant doesn't exist for the given ID") {
+      return res.status(404).send({ status: 404, message: "Applicant doesn't exist for the given ID" });
+    }
+    return res.status(500).send({ status: 500, message: "Insufficient Permissions" });
```

Comment:
Not sure if "Insufficient Permissions" makes too much sense here, the permissions check will throw HTTP 403 "unauthorized" if that is the case. I know you took this from analytics and tbh it doesn't make sense there either.

Change it to Server Error or something

Reply:
Sounds good,! Just pushed up the changes. Also you were right about the snapshot.data() not being the right format. I had to manually get all of the data and put it in an object. Should be good now.